src/lcm/dispatchers.py

```diff
@@ -12,64 +12,59 @@
-    that all combination variables must have the same size (e.g., in the simulation the
-    states act as combination variables, and their size equals the number of
-    simulations).
+    This function maps the function `func` over the simulation state-choice-space. That
+    is, it maps `func` over the Cartesian product of the choice variables, and over the
+    fixed simulation states. For each choice variable, a leading dimension is added to
+    the output object, with the length of the axis being the number of possible values
+    in the grid. Importantly, it does not create a Cartesian product over the state
+    variables, since these are fixed during the simulation. For the state variables,
+    a single dimension is added to the output object, with the length of the axis
+    being the number of simulated states.
 
-    spacemap preserves the function signature and allows the function to be called with
-    keyword arguments.
+    simulation_spacemap preserves the function signature and allows the function to be
+    called with keyword arguments.
 
     Args:
         func: The function to be dispatched.
-        product_vars: Names of the product variables, i.e. those that are stored as
+        choices_var_names: Names of the choice variables, i.e. those that are stored as
             arrays of possible values in the grid, over which we create a Cartesian
             product.
-        combination_vars: Names of the combination variables, i.e. those that are
-            stored as arrays of possible combinations.
+        states_var_names: Names of the state variables, i.e. those that are stored as
+            arrays of possible states.
 
     Returns:
         A callable with the same arguments as func (but with an additional leading
-        dimension) that returns a jax.Array or pytree of arrays. If `func` returns a
-        scalar, the dispatched function returns a jax.Array with k + 1 dimensions, where
-        k is the length of `product_vars` and the additional dimension corresponds to
-        the `combination_vars`. The order of the dimensions is determined by the order
-        of `product_vars`. If the output of `func` is a jax pytree, the usual jax
+        dimension) that returns an Array or pytree of Arrays. If `func` returns a
+        scalar, the dispatched function returns an Array with k + 1 dimensions, where k
+        is the length of `choices_var_names` and the additional dimension corresponds to
+        the `states_var_names`. The order of the dimensions is determined by the order
+        of `choices_var_names`. If the output of `func` is a jax pytree, the usual jax
         behavior applies, i.e. the leading dimensions of all arrays in the pytree are as
         described above but there might be additional dimensions.
 
     """
-    if duplicates := find_duplicates(product_vars, combination_vars):
+    if duplicates := find_duplicates(choices_var_names, states_var_names):
```

Comment:
IIUC, when running `solve_and_simulate`, this check would only be made after the first model solution. Can we do that upfront somehow and assume here that we get correct input? Not necessarily for this PR, ofc!

Reply:
This is checked when a user creates a `Model` specification instance. I like having this check here, so that it is easier for me to debug when I mess up something in between the user model and calling the dispatchers.

Reply:
Cool. No need to leave this open, then, just add a little comment in the code.